### ISSUE TITLE
[BUG] Incompatible with react 18

### ISSUE THREAD

**Issue by onionhammer** [microsoft/applicationinsights-react-js#7]
**Description/Screenshot**
```
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^17.0.1" from @microsoft/applicationinsights-react-js@3.2.4 
npm WARN   node_modules/@microsoft/applicationinsights-react-js
npm WARN     @microsoft/applicationinsights-react-js@"^3.2.4" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-number-format@4.9.1
npm WARN Found: react@18.0.0
npm WARN node_modules/react
npm WARN   react@"^18.0.0" from the root project
npm WARN   6 more (@testing-library/react, react-dom, @compiled/react, ...)
npm WARN 
```

**Steps to Reproduce**
1. Install latest version of react
2. Install @microsoft/applicationinsights-react-js@3.2.4

**Comment by LarsKemmann** [comment]
> As a general approach (mostly because of resourcing) we don't jump onto new versions of frameworks until it's been released for at least 6 months (as they _may_ fix breaking changes).

This is self-contradictory. On the one hand, it admits that the problem is a resource constraint on the team. On the other hand, it attempts to blame frameworks' potential instability for the decision.

Considering the size of React 18's userbase and the amount of time it's been in beta (and, at this point, out of beta), waiting "at least" 6 months is absurd. This is a Microsoft resource prioritization issue and needs to be addressed.

**Comment by MSNev** [comment]
#15

**Comment by MSNev** [comment]
We believe that this should be addressed simplistically by #15 above, this is now available as a [nightly](https://www.npmjs.com/package/@microsoft/applicationinsights-react-js/v/3.3.7-nightly.2208-05) package. If we could get some validation to confirm would be helpful.

**Comment by LarsKemmann** [comment]
Works for me @MSNev, thank you very much!! 🎉🚀
